### PR TITLE
MariaDB cluster needs to have (at least) 3 nodes.

### PR DESCRIPTION
**Pull request by jsuchome**
Which means that nothing (database/rabbitmq) could rely on DRBD
based cluster (which needs to be of size 2).

### CODE REVIEW

**Comment by jsuchome** [comment]
[P9 (5) Adapt mkcloud (and jenkins jobs) for MariaDB HA deployment](https://trello.com/c/oxcZyEgJ/5-p9-5-adapt-mkcloud-and-jenkins-jobs-for-mariadb-ha-deployment)

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -1979,7 +1979,9 @@ function hacloud_configure_cluster_defaults
 
 function hacloud_configure_data_cluster
 {
-    proposal_set_value pacemaker $clusternamedata "['attributes']['pacemaker']['drbd']['enabled']" true
+    if [[ "$want_database_sql_engine" -ne "mysql" ]] ; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: Integer comparison is wrong here, use !=

File: scripts/qa_crowbarsetup.sh [hunk 2]
```diff
@@ -2108,8 +2113,14 @@ function custom_configuration
         ;;
         rabbitmq)
             if [[ $hacloud = 1 ]] ; then
-                proposal_set_value rabbitmq default "['attributes']['rabbitmq']['ha']['storage']['mode']" "'drbd'"
-                proposal_set_value rabbitmq default "['attributes']['rabbitmq']['ha']['storage']['drbd']['size']" "$drbd_rabbitmq_size"
+                if [[ "$want_database_sql_engine" -eq "mysql" ]] ; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: Hmm, how does rabbitmq depend on the database backend? It doesn't - maybe this should be dependent on something else, like the cloudversion for example

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Oh

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: `$cloudversion` is not good, as we want this for SOC7 as well, and also currently support DRBD for rabbitmq. I know it's not ideal , but `want_database_sql_engine` seems the best one currently

File: scripts/qa_crowbarsetup.sh [hunk 3]
```diff
@@ -2108,8 +2113,14 @@ function custom_configuration
         ;;
         rabbitmq)
             if [[ $hacloud = 1 ]] ; then
-                proposal_set_value rabbitmq default "['attributes']['rabbitmq']['ha']['storage']['mode']" "'drbd'"
-                proposal_set_value rabbitmq default "['attributes']['rabbitmq']['ha']['storage']['drbd']['size']" "$drbd_rabbitmq_size"
+                if [[ "$want_database_sql_engine" == "mysql" ]] ; then
```
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: I believe its = not ==

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: can you add a fixme? this otherwise looks like a typo then.

File: scripts/qa_crowbarsetup.sh (hunk 3 above)
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: No, it is correct, I have it the other way in this block

File: scripts/qa_crowbarsetup.sh (hunk 2 above)
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: OK

File: scripts/qa_crowbarsetup.sh (hunk 3 above)
Comment (jsuchome) [inline scripts/qa_crowbarsetup.sh]: Ah, sorry, I misread it.
Anyway, I think both `=` and `==` are correct and we're using both in mkcloud code

Comment (nicolasbock) [inline scripts/qa_crowbarsetup.sh]: The bash builtin `[[ ]]` understands both `=` and `==`. The external `test` on the other hand needs a single `=`.

**Comment by jsuchome** [comment]
Cinder related failures should be fixed with https://github.com/crowbar/crowbar-openstack/pull/1157 (at least some of them)

**Comment by jsuchome** [comment]
Let's merge it and what the failures in relevant jenkins jobs

**Pull request merged**